shorebird_code_push/lib/src/shorebird_code_push.dart

```diff
@@ -2,48 +2,101 @@ import 'dart:isolate';
 
 import 'package:shorebird_code_push/src/updater.dart';
 
+/// A function that constructs an [Updater] instance. Used for testing.
+typedef UpdaterBuilder = Updater Function();
```

Comment:
You can use @visibleForTesting if you like, although that would add a package:meta dependency.

Reply:
We already had the meta dependency, so I did that and also made a `forTest` constructor so UpdaterBuilder isn't part of the public API.